tests/butlerProxy.py

```diff
@@ -41,16 +41,17 @@
 class ButlerProxyTestCase(unittest.TestCase):
     """A test case for the data butler finding a Mapper in a root"""
 
-    def setUp(self):
-        if os.path.exists(os.path.join(ROOT, 'root/proxyOut')):
-            shutil.rmtree(os.path.join(ROOT, 'root/proxyOut'))
+    inputDir = os.path.join(ROOT, 'root')
+    outputDir = os.path.join(ROOT, 'ButlerProxyTestCase')
 
-        self.butler = dafPersist.Butler(os.path.join(ROOT, "root"), outPath="proxyOut")
+    def setUp(self):
+        self.tearDown()
+        self.butler = dafPersist.Butler(self.inputDir,
+                                        outPath=os.path.join(self.outputDir, "proxyOut"))
 
     def tearDown(self):
-        del self.butler
-        if os.path.exists(os.path.join(ROOT, 'root/out')):
-            shutil.rmtree(os.path.join(ROOT, 'root/out'))
+        if os.path.exists(self.outputDir):
+            shutil.rmtree(self.outputDir)
```

Comment:
Just to clarify: it is definitely true that we must explicitly delete test case attributes that inherit from Citizen in `tearDown`. If we don't, the memory test fails. Allowing such objects to be garbage collected definitely does not work. It is not a question of whether the objects eventually go out of scope -- of course they do -- it's whether they go out of scope fast enough that the memory test doesn't think memory has been leaked.

Also, I thought CPython did delete immediately when the ref count dropped to 0 (unless there are cycles, which take longer to clean up?) -- though I agree the language specification does not require that. If this was not the case, then I'm surprised our memory test is usable; I would expect random failures in most of our packages.